File: daq/slc/dqm/src/Histo1I.cc
#include "daq/slc/dqm/Histo1I.h"

#include "daq/slc/dqm/IntArray.h"

using namespace Belle2;

Histo1I::Histo1I() throw() : Histo1()
{
  m_data = new IntArray(0);
}

Histo1I::Histo1I(const Histo1I& h) throw() : Histo1(h)
{
  m_data = new IntArray(h.m_data->size());
}

Histo1I::Histo1I(const std::string& name, const std::string& title, int nbinx, double min, double max)
throw() : Histo1(name, title, nbinx, min, max)
{
  m_data = new IntArray(nbinx + 2);
}

Histo1I::~Histo1I() throw() {}

std::string Histo1I::getDataType() const throw()
{
  return std::string("H1I");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/Histo1I.java
package b2daq.dqm.core;


public class Histo1I extends Histo1 {

	public Histo1I() {
		super();
		data = new IntArray(0);
	}
	public Histo1I(String name, String title, int nbinx, double xmin, double xmax) {
		super(name, title, nbinx, xmin, xmax);
		data = new IntArray(nbinx+2);
	}
	public Histo1I(Histo1I h) {
		super();
		data = new IntArray(0);
	}
	public String getDataType() {
		return "H1I";
	}
	
}
